clients/src/main/java/org/apache/kafka/common/security/kerberos/Login.java

```diff
@@ -286,11 +286,12 @@ public Subject subject() {
     private synchronized LoginContext login(final String loginContextName) throws LoginException {
         String jaasConfigFile = System.getProperty(JaasUtils.JAVA_LOGIN_CONFIG_PARAM);
         if (jaasConfigFile == null) {
-            throw new IllegalArgumentException("You must pass " + JaasUtils.JAVA_LOGIN_CONFIG_PARAM + " in secure mode.");
+            log.debug("System property '" + JaasUtils.JAVA_LOGIN_CONFIG_PARAM + "' is not set, using default JAAS configuration.");
```

Comment:
I'm wondering why you changed from throwing an illegal argument exception to just logging a debug message. Don't you want the error to propagate up the call stack?